extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/MicroProfileRestClientRequestFilter.java

```diff
@@ -63,7 +65,21 @@ public void filter(ClientRequestContext requestContext) {
         }
 
         if (clientHeadersFactory != null) {
-            incomingHeaders = clientHeadersFactory.update(incomingHeaders, headers);
+            if (clientHeadersFactory instanceof ReactiveClientHeadersFactory
+                    && requestContext instanceof ResteasyReactiveClientRequestContext) {
```

Comment:
this filter should always work with `RestEasyReactiveClientRequestContext`, no? I don't think we need to check it with `instanceof`

Reply:
I think so... but I thought we could be extra safe here. However, will remove the instanceof.